pkg/image/airgap_test.go

```diff
@@ -0,0 +1,143 @@
+package image
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"testing"
+
+	dockerregistrytypes "github.com/replicatedhq/kots/pkg/docker/registry/types"
+	"github.com/replicatedhq/kots/pkg/image/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPushEmbeddedClusterArtifacts(t *testing.T) {
+	tests := []struct {
+		name          string
+		airgapFiles   map[string][]byte
+		wantArtifacts map[string]string
+		wantErr       bool
+	}{
+		{
+			name: "no embedded cluster files",
+			airgapFiles: map[string][]byte{
+				"airgap.yaml":      []byte("this-is-the-airgap-metadata"),
+				"app.tar.gz":       []byte("this-is-the-app-archive"),
+				"images/something": []byte("this-is-an-image"),
+			},
+			wantArtifacts: map[string]string{},
+			wantErr:       false,
+		},
+		{
+			name: "has embedded cluster files",
+			airgapFiles: map[string][]byte{
+				"airgap.yaml":                       []byte("this-is-the-airgap-metadata"),
+				"app.tar.gz":                        []byte("this-is-the-app-archive"),
+				"images/something":                  []byte("this-is-an-image"),
+				"embedded-cluster/test-app":         []byte("this-is-the-binary"),
+				"embedded-cluster/charts.tar.gz":    []byte("this-is-the-charts-bundle"),
+				"embedded-cluster/images-amd64.tar": []byte("this-is-the-images-bundle"),
+				"embedded-cluster/some-file-TBD":    []byte("this-is-an-arbitrary-file"),
+			},
+			wantArtifacts: map[string]string{
```

Comment:
would be great to test the actual path in the registry including the defined registry namespace and `embedded-cluster`. would that be difficult to add?

Reply:
i think that would just be a small change in the test. let me see...